content/en/blog/_posts/2025-02-14-cloud-controller-manager-chicken-egg-problem.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: blog
+title: "The Cloud Controller Manager Chicken and Egg Problem"
+date: 2025-02-14
+slug: cloud-controller-manager-chicken-egg-problem
+author: >
+  Antonio Ojea,
+  Michael McCune
+draft: true
+---
+
+Kubernetes 1.31
+[completed the largest migration in Kubernetes history][migration-blog], removing the in-tree
+cloud provider.  While the component migration is now done, this leaves some additional
+complexity for users and installer projects (for example, kOps or Cluster API) .  We will go
+over those additional steps and failure points and make recommendations for cluster owners.
+This migration was complex and some logic had to be extracted from the core components,
+building four new subsystems.
+
+1. **Cloud controller manager** ([KEP-2392][kep2392])
+2. **API server network proxy** ([KEP-1281][kep1281])
+3. **kubelet credential provider plugins** ([KEP-2133][kep2133])
+4. **Storage migration to use [CSI][csi]** ([KEP-625][kep625])
+
+The [cloud controller manager is part of the control plane][ccm]. It is a critical component
+that replaces some functionality that existed previously in the kube-controller-manager and the
+kubelet.
+
+![Components of Kubernetes](https://kubernetes.io/images/docs/components-of-kubernetes.svg)
```

Comment:
nit:
- should use a `figure` shortcode
- should use a site-relative link

We can fix these post-publication.